lib/clarivate/links_client.rb

```diff
@@ -0,0 +1,65 @@
+module Clarivate
+  # Links AMR (Article Match Retrieval) Service
+  # @see http://ipscience-help.thomsonreuters.com/LAMRService/WebServicesOverviewGroup/overview.html Service documentation
+  class LinksClient
+    attr_reader :username, :password, :host
+
+    def initialize(username: nil, password: nil, host: 'https://ws.isiknowledge.com')
+      @host = host
+      @username = username
+      @password = password
+      (@username, @password) = Base64.decode64(Settings.WOS.AUTH_CODE).split(':', 2) unless username
+    end
+
+    # limited to 50 ids
+    # @param [Array<String>] ids
+    # @return [Hash<String => Hash>]
+    def links(ids)
+      raise ArgumentError, "1-50 ids required" unless ids.present? && ids.count <= 50
+      response = connection.send(:post) do |req|
+        req.path = '/cps/xrpc'
+        req.body = request_body(ids.uniq)
+      end
+      ng = Nokogiri::XML(response.body) { |config| config.strict.noblanks }.remove_namespaces!
+      pairs = ng.xpath('response/fn/map/map').map do |node|
+        [node.attr('name').split('_').last, vals_to_hash(node.xpath('//val'))]
+      end
+      pairs.to_h
+    end
+
+    private
+
+      # @return [Faraday::Connection]
+      def connection
+        @connection ||= Faraday.new(url: host) do |faraday|
+          faraday.use Faraday::Response::RaiseError
+          faraday.adapter :httpclient
+        end
+      end
```

Comment:
Add a client logger in case we need to debug failures.

Reply:
If you need to debug, override `@connection` or do it in Faraday middleware.

Reply:
See the `NotificationManager` for the patterns established for this app.

Reply:
There are plenty of trash patterns in the app to be avoided.

Reply:
Integration with https://github.com/sul-dlss/sul_pub/blob/master/lib/notification_manager.rb will be required to merge any API client.

Reply:
That won't be happening in this PR.  For one thing the use pattern is completely different: this client is *ancillary*, for data enhancement.  Right now, we project the initial version of the links client will only be executing inside the WoS client, that already has its own logging.  The case for a separate log has not been made.

Secondly, even if we determined we wanted additional logging infrastructure, it certainly isn't required *now*.  This provides a baseline for other work to proceed.  It isn't supposed to be a complete integration, based on the scope of tickets represented.